### PR TITLE
Second module completed

### PR DESCRIPTION
**Pull request by JavohirMirzaakbarov**

### CODE REVIEW

File: NodeJS HW2/index.js [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    res.status(404).json({ message: `User with id=${id} was not found!` });
+  else {
+    res.locals.user = user;
+    next();
+  }
+});
+
+router.get("/", (req, res) => {
+  const users = getExistingUsers();
+  const { limit, loginSubstring } = req.body;
```
Comment (reactuz) [inline NodeJS HW2/index.js]: it's not a good practice to include body in GET request, plz consider other alternatives

File: NodeJS HW2/index.js [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    id: uuidv4(),
+    isDeleteted: false,
+  };
+  users.push(newUser);
+  res
+    .status(201)
+    .json({ message: "New user created successfully!", data: newUser });
+});
+
+router.patch("/:id", (req, res) => {
```
Comment (reactuz) [inline NodeJS HW2/index.js]: you also need validation here

File: NodeJS HW2/index.js (hunk 1 above)
Comment (JavokhirbekMirzaakbarov) [reply NodeJS HW2/index.js]: I will use query params then.

File: NodeJS HW2/index.js (hunk 2 above)
Comment (JavokhirbekMirzaakbarov) [reply NodeJS HW2/index.js]: Will add

**Pull request merged**